packages/compass-aggregations/src/components/stage-preview.tsx

```diff
@@ -0,0 +1,399 @@
+
+import type { RootState } from '../modules';
+import { gotoOutResults } from '../modules/out-results-fn';
+import { runStage } from '../modules/pipeline-builder/stage-editor';
+import {
+  isMissingAtlasStageSupport,
+  MERGE_STAGE_PREVIEW_TEXT,
+  OUT_STAGE_PREVIEW_TEXT,
+} from '../utils/stage';
+
+import LoadingOverlay from './loading-overlay';
+import { AtlasStagePreview } from './atlas-stage-preview';
+
+const stagePreviewOutStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'center',
+  width: '100%',
+  paddingTop: spacing[5],
+  paddingLeft: spacing[1],
+  paddingRight: spacing[1],
+  paddingBottom: spacing[2]
+});
+
+const stagePreviewOutTextStyles = css({
+  padding: `0 ${spacing[3]}px ${spacing[1]}px ${spacing[3]}px`,
+  textAlign: 'center',
+
+  // TODO: is this still used?
+  '&:not(:last-child)': {
+    paddingBottom: '8px'
+  }
+});
+
+// TODO: double-check that this is actually needed
+const stagePreviewOutLinkStyles = css({
+  border: 'none',
+  padding: 0,
+  margin: 0,
+  background: 'none'
+});
+
+type MergeProps = {
+  isComplete: boolean,
+  hasServerError: boolean,
+  isAtlasDeployed: boolean,
+  gotoMergeResults: () => void,
+  saveDocuments: () => void
+};
+
+function MergeSection({
+  isComplete,
+  hasServerError,
+  isAtlasDeployed,
+  gotoMergeResults,
+  saveDocuments
+}: MergeProps) {
+  if (isComplete) {
+    if (hasServerError) {
+      return (<div className={stagePreviewOutStyles} />);
+    }
+
+    // TODO: is it still possible to get here? How do we test this?
+    return (
+      <div className={stagePreviewOutStyles}>
+        <Body className={stagePreviewOutTextStyles}>
+          Documents persisted to collection specified by $merge.
+        </Body>
+        <Link
+          data-testid="go-to-merge-collection"
+          as="button"
+          className={stagePreviewOutLinkStyles}
+          onClick={gotoMergeResults}
+        >
+          Go to collection.
+        </Link>
+      </div>
+    );
+  }
+
+  return (
+    <div className={stagePreviewOutStyles}>
+      <div className={stagePreviewOutTextStyles}>
+        {MERGE_STAGE_PREVIEW_TEXT}
+      </div>
+      {isAtlasDeployed && (
+        <Button
+          variant="primary"
+          data-testid="save-merge-documents"
+          onClick={saveDocuments}
+        >
+          Merge documents
+        </Button>
+      )}
+    </div>
+  );
+}
+
+type OutProps = {
+  isComplete: boolean,
+  hasServerError: boolean,
+  isAtlasDeployed: boolean,
+  gotoOutResults: () => void,
+  saveDocuments: () => void,
+  stageValue: string
+};
+
+function OutSection({
+  isComplete,
+  hasServerError,
+  isAtlasDeployed,
+  gotoOutResults,
+  saveDocuments,
+  stageValue
+}: OutProps) {
+    if (isComplete) {
+      if (hasServerError) {
+        return (<div className={stagePreviewOutStyles} />);
+      }
+
+      // TODO: is it still possible to get here? How do we test this?
+      return (
```

Comment:
Yes, for both the stages `$out/$merge`, but only on cloud. You can simple test is by setting atlas in the state.